Sources/Build/BuildDelegate.swift

```diff
@@ -553,18 +553,22 @@ public final class BuildDelegate: BuildSystemDelegate, SwiftCompilerOutputParser
             }
 
             if let output = message.standardOutput {
-                // Scoop out any errors from the output, so they can later be passed to the advice provider in case of failure.
-                let regex = try! RegEx(pattern: #".*(error:[^\n]*)\n.*"#, options: .dotMatchesLineSeparators)
-                for match in regex.matchGroups(in: output) {
-                    self.errorMessagesByTarget[parser.targetName] = (self.errorMessagesByTarget[parser.targetName] ?? []) + [match[0]]
-                }
-                
+                // first we want to print the output so users have it handy
                 if !self.isVerbose {
                     self.progressAnimation.clear()
                 }
 
                 self.outputStream <<< output
                 self.outputStream.flush()
+
+                // next we want to try and scoop out any errors from the output (if reasonable size, otherwise this will be very slow),
+                // so they can later be passed to the advice provider in case of failure.
+                if output.utf8.count < 1024 * 10 {
```

Comment:
Is the 8 MB limit calculated in some way, or is it arbitrarily chosen? I don't have an opinion either way. I'm just curious.

Reply:
base don experiments with SIL output in crashes. @abertelrud is working on a separate PR to fix this parsing logic and this will be removed then. this PR is focused on "damage control" given that this caused issues in 5.4